grpcserver/server.go

```diff
@@ -56,37 +60,50 @@ type shipperServer struct {
 	ctx   context.Context
 	stop  func()
 
-	cfg Config
+	strictMode bool
 
 	pb.UnimplementedProducerServer
 }
 
 // NewShipperServer creates a new server instance for handling gRPC endpoints.
-func NewShipperServer(cfg Config, publisher Publisher) (ShipperServer, error) {
+// publisher can be set to nil, in which case the SetOutput() method must be called.
+func NewShipperServer(publisher Publisher) (ShipperServer, error) {
+	log := logp.NewLogger("shipper-server")
 	if publisher == nil {
-		return nil, errors.New("publisher cannot be nil")
+		log.Debugf("gRPC endpoint has no output, will wait for config")
 	}
 
 	id, err := uuid.NewV4()
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("error generating shipper UUID: %w", err)
 	}
 
 	s := shipperServer{
-		uuid:      id.String(),
-		logger:    logp.NewLogger("shipper-server"),
-		publisher: publisher,
-		close:     &sync.Once{},
-		cfg:       cfg,
+		uuid:       id.String(),
+		logger:     log,
+		publisher:  publisher,
+		close:      &sync.Once{},
+		strictMode: false,
 	}
 
 	s.ctx, s.stop = context.WithCancel(context.Background())
 
 	return &s, nil
 }
 
+// SetStrictMode updates the strict mode setting for the server
+// A bit of a hack, but needed as this value comes from the output unit, not the input with the rest of the gRPC config.
+// TODO: can we make a change to get this from the input unit instead?
```

Comment:
Why would this help?

Reply:
The comment was before I realized that was a development flag and I was still trying to hack it to make it work, probably not needed.